### PR TITLE
[WIP] Add Turbo Compass.

### PR DESCRIPTION
**Pull request by sprillo**

### CODE REVIEW

File: compass/turbo_compass.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+            cv_model=lambda state:
+                TrainValSplitCVMatrixCompletionModel(
+                    ExcludeConstantColumnsModelWrapper(
+                        ColumnNormalizerModelWrapper(
+                            MatrixCompletionFastALS(
+                                n_factors=max(1, int(min(state.R, state.C) * state.sampled_density * 0.5)),
+                                lam=10.0,
+                                n_epochs=100,
+                                verbose=False))),
+                    train_ratio=0.8,
```
Comment (sprillo) [inline compass/turbo_compass.py:324]: One simple improvement after discussing with Richard Zhang is to make the size of the cross-validation set adaptive:
- When we start off we have little data, so maybe a 80/20 split produces too small a validation set size.
- Once we have more data, a 80/20 split might be too big, i.e. we are being too conservative as the algorithm progresses (because we shrink the training set size more than necessary), and as a consequence calling Compass more than we should to meet the user's specified reconstruction quality.

Comment (sprillo) [reply compass/turbo_compass.py:324]: It sounds to me that what would be ideal would be to use a fixed CV size for each reaction, say size 50. In that case I would need to implement another concretion for `CVMatrixCompletionModel` which performs CV splitting based on a fixed CV size, rather than on a percentage.

Comment (sprillo) [reply compass/turbo_compass.py:324]: (or, rather than creating a new concretion, cleverly factor out the CV-splitting behavior to get the new behavior via composition)

Comment (sprillo) [reply compass/turbo_compass.py:324]: Or maybe, it's easier to just add a `max_cv_size` argument to the `TrainValSplitCVMatrixCompletionModel` API.
